daemon/image_history.go

```diff
@@ -31,7 +32,9 @@ func (daemon *Daemon) ImageHistory(name string) ([]*image.HistoryResponseItem, e
 			if len(img.RootFS.DiffIDs) <= layerCounter {
 				return nil, fmt.Errorf("too many non-empty layers in History section")
 			}
-
+			if !system.IsOSSupported(img.OperatingSystem()) {
+				return nil, system.ErrNotSupportedOperatingSystem
+			}
```

Comment:
How does this relate to the rest of the PR?

Reply:
It's a safety check to ensure that we don't panic on L39 where the layerstores array is indexed by the operating system. Strictly it's not related to this PR, it should have been in https://github.com/moby/moby/pull/34859, for which this PR is a direct follow up. Happy to move it out to a separate PR if you prefer.